sim-rs/LINEAR_LEIOS.md

```diff
@@ -0,0 +1,30 @@
+# Linear Leios (rust simulation)
+
+To run it, set `leios-variant` to `linear`.
+
+The log file schema is currently identical to every other variant (though `pipeline` is always 0).
+
+## Description
+
+Whenever a node creates an RB, it also creates an EB. The RB header contains a reference to this new EB. If the RB producer has a certificate for the parent RB’s EB, it will include that certificate in the RB body.
+
+RB headers are diffused separately from bodies. When a node receives an RB header, it checks whether that RB should be the new head of its chain. If so, it will request the RB body and the referenced EB (from the first peer which announces them).
+
+When a node receives an RB body, it immediately removes all referenced/conflicting transactions from its mempool. If the RB has an EB certificate, it also removes that EB’s transactions from its mempool. 
```

Comment:
> `If the RB has an EB certificate, it also removes that EB’s transactions from its mempool.`

When the node issues an RB, which txs does it include? If the above rule is the only way EBs evict txs from the mempool, then I'm suspicious that the RB duplicates a prefix of its parent's EB.

Reply:
@nfrisby good callout, this mempool pruning is actually redundant. If the parent EB was produced >= `L_vote + L_diff` slots ago and certified in time, the node would already have evicted its TXs from the mempool..